modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/display/context/ContentPageEditorDisplayContext.java

```diff
@@ -554,7 +558,8 @@ public Map<String, Object> getEditorContext(String npmResolvedPackageName)
 					"/layout_content_page_editor/get_page_preview",
 					_isPreviewPageAsGuestUser())
 			).put(
-				"publishURL", getPublishURL()
+				"publishURL",
+				_isAVariantOfATest() ? getSaveVariantURL() : getPublishURL()
```

Comment:
We can't use a ternary expression following Liferay Guidelines :(.

But what do you think if we have two different urls?

once the publishURL and another the saveVariantURL?,

It implies a change in the frontend, but maybe it is clearer that way

Reply:
Hey @cgoncas ,

ok, I will change the ternary expression. 

No, it does not imply a frontend change because now we are using the same url to save the variant and to publish the page but the text button changes, maybe this is not right but it is implemented like this.